src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -227,6 +227,18 @@ public void writeTo(DownloadInstance di, Class<?> clazz, Type type, Annotation[]
                         // (similarly to what the Access API returns when a thumbnail is requested on a text file, etc.)
                         throw new NotFoundException("datafile access error: requested optional service (image scaling, format conversion, etc.) could not be performed on this datafile.");
                     }
+                } else if (di.getAuxiliaryFile() != null) {
+                    String auxTag = di.getAuxiliaryFile().getFormatTag(); 
+                    String auxVersion = di.getAuxiliaryFile().getFormatVersion();
+                    if (auxVersion != null) {
+                        auxTag = auxTag + "_" + auxVersion;
+                    }
+                    long auxFileSize = di.getAuxiliaryFile().getFileSize();
+                    InputStreamIO auxStreamIO = new InputStreamIO(storageIO.getAuxFileAsInputStream(auxTag), auxFileSize);
+                    auxStreamIO.setFileName(storageIO.getFileName() + "." + auxTag);
+                    auxStreamIO.setMimeType(di.getAuxiliaryFile().getContentType());
+                    storageIO = auxStreamIO;
```

Comment:
storageIO.open called above creates the inputStream for the main file. If it's not going to be used, it needs to be closed (storageIO.getInpputStream().close() - StorageIO doesn't implement Closeable yet/have a close() method - which could be useful someday).

Reply:
Thanks, I missed that. Fixed. 
It's not pretty, what goes on in that DownloadInstance/DownloadInstanceWriter setup, btw. I am opening an issue for cleaning up/simplifying/rewriting it.